### ISSUE TITLE
llama-2-7b-chat部署后乱码

### ISSUE THREAD

**Issue by Sun-9923** [InternLM/xtuner#126]
1. 使用`xtuner train llama2_7b_chat_qlora_alpaca_zh_e3`在4090上训练了1个epoch
2. 使用`xtuner convert pth_to_hf ${CONFIG_NAME_OR_PATH} ${PTH} ${SAVE_PATH}`命令将训练的adapter转换为了hf格式
3. 使用`xtuner chat ${NAME_OR_PATH_TO_LLM} --adapter [{NAME_OR_PATH_TO_ADAPTER}`命令测试，英文对话无乱码，中文对话几乎全是乱码
<img width="1259" alt="iShot_2023-09-09_10 18 28" src="https://github.com/InternLM/xtuner/assets/77619190/5fae2ef6-3ff2-4d7b-b076-6a0cc905dc57">

4. 使用`xtuner chat models--meta-llama--Llama-2-7b-chat-hf/snapshots/08751db2aca9bf2f7f80d2e516117a53d7450235 --adapter llama2-7b-hf --no-streamer`测试时中文乱码问题解决。
5. 使用`xtuner convert merge \
    ${NAME_OR_PATH_TO_LLM} \
    ${NAME_OR_PATH_TO_ADAPTER} \
    ${SAVE_PATH} `合并模型
6. 使用`pip install lmdeploy
python -m lmdeploy.pytorch.chat ${NAME_OR_PATH_TO_LLM} \
    --max_new_tokens 256 \
    --temperture 0.8 \
    --top_p 0.95 \
    --seed 0`部署后中英文皆有乱码
<img width="1258" alt="iShot_2023-09-15_11 04 14" src="https://github.com/InternLM/xtuner/assets/77619190/55fac9a8-f6f9-4df8-b4c0-4865086e14bc">

7. 使用`python -m lmdeploy.pytorch.chat llama2-7b-deploy --max_new_tokens 256     --temperture 0.8     --top_p 0.95     --seed 0     --no-streamer`重新部署后仍有乱码
<img width="1258" alt="iShot_2023-09-15_11 05 35" src="https://github.com/InternLM/xtuner/assets/77619190/3ab40e50-2218-482b-944d-41e95de8fe68">

**Comment by maxchiron** [comment]
```xtuner chat``` keeps the history chats. So please try reload the fine-tuned model and only input english content and see what happened? ```--no-streamer --prompt-template``` of course

**Comment by Sun-9923** [comment]
> `xtuner chat` keeps the history chats. So please try reload the fine-tuned model and only input english content and see what happened? `--no-streamer --prompt-template` of course

Thank you for your response. The garbled text issue in `xtuner chat` was resolved after I added `--no-streamer`. However, when deploying with lmdeploy, the garbled text issue still persists even with the `--no-streamer ` option added.

**Comment by LZHgrla** [comment]
lmdeploy 的 pytorch 推理引擎现阶段不支持 `--no-streamer` 参数，会强制使用 `transformers` 的 streamer 接口，这会导致中文出现乱码。

如果确需使用 lmdeploy，可以考虑使用 lmdeploy的 turbomind 推理引擎，命令如下：
```python
python3 -m lmdeploy.serve.turbomind.deploy llama2 ${LLAMA2_LLM} --dst_path ${SAVE_PATH}
python -m lmdeploy.turbomind.chat ${SAVE_PATH}
```

如有其他关于 lmdeploy 的使用问题，欢迎在 lmdeploy 的 [issues](https://github.com/InternLM/lmdeploy/issues) 直接提问～

**Comment by Sun-9923** [comment]
@LZHgrla 
哦～原来是这样，感谢🙏
